nilearn/glm/_base.py

```diff
@@ -220,19 +220,17 @@ def generate_report(
             :class:`~nilearn.glm.second_level.SecondLevelModel` through
             :meth:`~nilearn.glm.second_level.SecondLevelModel.compute_contrast`.
 
-        first_level_contrast : :obj:`str` or :class:`numpy.ndarray` of \
-                            shape (n_col) with respect to \
-                            :class:`~nilearn.glm.first_level.FirstLevelModel` \
-                            or None, default=None
+        %(first_level_contrast)s
+
             For :class:`~nilearn.glm.second_level.SecondLevelModel`,
             in case a :obj:`list` of
-            :class:`~nilearn.glm.first_level.FirstLevelModel` was provided
-            as ``second_level_input``,
+        :class:`~nilearn.glm.first_level.FirstLevelModel`
+        or a dataframe was provided as ``second_level_input``,
             we have to provide a :term:`contrast`
             to apply to the first level models
             to get the corresponding list of images desired,
             that would be tested at the second level.
```

Comment:
The same here.

Reply:
not quite 

first_level_contrast in _utils.docs is the thing that is relevant only for SecondLevelModel that you will find in nilear/glm/second_level/second_level.py.

but for reports, the same API is used for both first and second level, so I feel that an extra explanation is needed to say why this parameter is necessary for second level model but not first level

Reply:
actually when rereading the rendered doc it was a bit weird to see some parameters in FirstLevelModel.generate_report() that do not explain they have to be ignored so I have put everything in _utils.doc